### PR TITLE
Remove hard-coding of RAPIDS version where possible

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
## Description
* Read `VERSION` file from CMake
* Read `rmm.__version__` from docs build
* Read `VERSION` file from shell scripts
* Remove updates from `ci/release/update-version.sh`

Issue: https://github.com/rapidsai/build-planning/issues/15

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Kyle! 🙏

Think we can also make these changes: https://github.com/rapidsai/build-planning/issues/15#issuecomment-1960541988 + https://github.com/rapidsai/build-planning/issues/15#issuecomment-1960595970

File: ci/build_docs.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ rapids-mamba-retry install \
   --channel "${PYTHON_CHANNEL}" \
   rmm librmm
 
-export RAPIDS_VERSION_NUMBER="24.04"
+export RAPIDS_VERSION_NUMBER="$(sed -E -e 's/^([0-9]{2})\.([0-9]{2})\.([0-9]{2})/\1.\2/' VERSION)"
```
Comment (harrism) [inline ci/build_docs.sh:28]: Since this little sed script is used more than once, could it be put somewhere as a utility and documented?

Comment (KyleFromNVIDIA) [reply ci/build_docs.sh:28]: Opened https://github.com/rapidsai/gha-tools/pull/97.

Comment (harrism) [reply ci/build_docs.sh:28]: Awesome. I'll approve assuming you will merge this after adding calls to that script.

**Review by harrism — APPROVED** [review]
Thanks!

**Review by bdice — APPROVED** [review]

File: scripts/doxygen.sh [hunk 2]
```diff
@@ -21,6 +21,9 @@ if [ ! $(version "$DOXYGEN_VERSION") -eq $(version "1.9.1") ] ; then
   exit 0
 fi
 
+export RAPIDS_VERSION="$(sed -E -e 's/^([0-9]{2})\.([0-9]{2})\.([0-9]{2}).*$/\1.\2.\3/')"
+export RAPIDS_VERSION_MAJOR_MINOR="$(sed -E -e 's/^([0-9]{2})\.([0-9]{2})\.([0-9]{2}).*$/\1.\2/')"
```
Comment (bdice) [inline scripts/doxygen.sh:25]: Were these sed commands supposed to be replaced by `rapids-version` and `rapids-version-major-minor`?

Comment (KyleFromNVIDIA) [reply scripts/doxygen.sh:25]: No. Because this file can be run outside of CI, it can't rely on `gha-tools`.

Comment (KyleFromNVIDIA) [reply scripts/doxygen.sh]: Though I just noticed the `sed` commands are not actually reading from `VERSION`. Fixed.

Comment (bdice) [reply scripts/doxygen.sh]: Great! Okay, that part about `VERSION` was my second question :)

**Review by AyodeAwe — APPROVED** [review]

**Pull request merged**